python_modules/dagster/dagster/_core/snap/job_snapshot.py

```diff
@@ -157,17 +153,17 @@ def from_job_def(cls, job_def: JobDefinition) -> "JobSnap":
         lineage = None
         if job_def.op_selection_data:
             lineage = JobLineageSnap(
-                parent_snapshot_id=create_job_snapshot_id(
-                    cls.from_job_def(job_def.op_selection_data.parent_job_def)
-                ),
+                parent_snapshot_id=cls.from_job_def(
+                    job_def.op_selection_data.parent_job_def
+                ).snapshot_id,
                 op_selection=sorted(job_def.op_selection_data.op_selection),
                 resolved_op_selection=job_def.op_selection_data.resolved_op_selection,
             )
         if job_def.asset_selection_data:
             lineage = JobLineageSnap(
-                parent_snapshot_id=create_job_snapshot_id(
-                    cls.from_job_def(job_def.asset_selection_data.parent_job_def)
-                ),
+                parent_snapshot_id=cls.from_job_def(
+                    job_def.asset_selection_data.parent_job_def
+                ).snapshot_id,
```

Comment:
theres existing code over here https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_core/definitions/job_definition.py#L1027-L1046 that we should probably re-use that hits memoization so we hit the same parent snap & snapshot id